### PR TITLE
Refactor `HolidayBase::get_named`

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change
The PR aims to impove `HolidayBase::get_named` flexibility in terms of available holiday name lookup types:
  - Add `lookup` parameter.
  - Add `contains/iexact/icontains/iexact` lookup types.
  - Keep the current behavior with `icontains` as default lookup type.

## Type of change

- [ ] New country holidays support (thank you!)
- [ ] Supported country holidays update (calendar discrepancy fix, localization)
- [x] Existing code/tests/processes improvement (best practice, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (adds functionality to python-holidays in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts
- [x] The code style looks good (`make pre-commit`)
- [x] I've added tests to verify that the new code works and all tests pass locally (`make test`)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by mborsetti** [comment]
What's the motivation behind this PR? 

- I am having a hard time imagining the use case requiring using case-sensitivity when searching for a holiday;
- I can sort-of-see a use for regex, e.g. to find an observed Founder's Day holiday, although even this could be achieved easily by using ``get_named("founder's day")`` and applying a list comprehension on the result to remove the returned holidays that don't have ``observed`` in their name.

In any case, I would consider **not** replacing the default logic from one using ``str in`` (which is highly performant) to one based on ``re`` one (which isn't).

**Comment by KJhellico** [comment]
Somewhere in the tests I saw cases where the ability to check for **exact name** would be useful. But I agree that this is not a regular occurrence, so we need to consider performance.

**Comment by arkid15r** [comment]
As was pointed out by @mborsetti there're no need in regex based logic for these cases.
I've changed the implementation based on the review. Also changed the common.TestCase holiday name related assertions to use exact lookup.

PTAL

File: tests/countries/test_chile.py [hunk 1]
```diff
@@ -270,12 +270,14 @@ def test_columbus_day(self):
         )
 
         self.assertHolidayName(
```
Comment (KJhellico) [inline tests/countries/test_chile.py:272]: No need to change this test, in #990 I've updated this holiday name, it was wrong :)

Comment (arkid15r) [reply tests/countries/test_chile.py:272]: I believe I'll have to resolve this when I merge it :)

**Review by KJhellico — COMMENTED** [review]
LGTM

**Pull request merged**